src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +51,157 @@ const ActorInit Shot_Sun_InitVars = {
+                this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, fairyParams);
+
+    Actor_Kill(&this->actor);
+}
+
+void ShotSun_StartFairyCountdown(ShotSun* this, GlobalContext* globalCtx) {
+    if ((func_8005B198() == this->actor.type) || (this->timer != 0)) {
+        this->actionFunc = ShotSun_FairyCountdown;
+        this->timer = 50;
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_KANKYO, this->actor.initPosRot.pos.x,
+                    this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, 0x11);
+
+        func_80078914(&this->actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    }
+}
+
+void func_80BADF0C(ShotSun* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s32 params = this->actor.params & 0xFF;
+
+    if (func_800CB650(&this->actor.posRot.pos, &player->actor.posRot.pos) > 22500.0f) {
+        this->unk_1A4[0] = 0;
+    } else {
+        if (this->unk_1A4[0] == 0) {
+            if (!(player->stateFlags2 & 0x1000000)) {
+                player->stateFlags2 |= 0x800000;
+                return;
+            } else {
+                this->unk_1A4[0] = 1;
+            }
+        }
+        if (this->unk_1A4[0] == 1) {
+            func_8010BD58(globalCtx, 1U);
+            this->unk_1A4[0] = 2;
+            return;
+        }
+
+        if (this->unk_1A4[0] == 2 && globalCtx->msgCtx.unk_E3EE == 4) {
+            if (((params == 0x40) && (*(u16*)(&globalCtx->msgCtx.unk_E3E8[4]) == 9)) ||
+                ((params == 0x41) && (*(u16*)(&globalCtx->msgCtx.unk_E3E8[4]) == 0xB))) {
+                this->actionFunc = ShotSun_StartFairyCountdown;
+                func_80080480(globalCtx, this);
+                this->timer = 0;
+            } else {
+                this->unk_1A4[0] = 0;
+            }
+            this->unk_1A4[0] = 0;
+        }
+    }
+}
+
+void ShotSun_HyliaSunUpdate(ShotSun* this, GlobalContext* globalCtx) {
+    Vec3s cylinderPos;
+    Player* player = PLAYER;
+    EnItem00* collectible;
+    s32 pad;
+    Vec3f spawnPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/func_80BADDCC.s")
+    if ((this->collider.base.acFlags & 2) != 0) {
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+        osSyncPrintf(VT_FGCOL(CYAN) "SHOT_SUN HIT!!!!!!!\n" VT_RST);
+        if (INV_CONTENT(SLOT_ARROW_FIRE) == ITEM_NONE) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_ETCETERA, 700.0f, -800.0f, 7261.0f, 0, 0, 0, 7);
+            globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02007020);
+            if (1) {}
+            gSaveContext.cutsceneTrigger = 1;
+        } else {
+            spawnPos.x = 700.0f;
+            spawnPos.y = -800.0f;
+            spawnPos.z = 7261.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/func_80BADE74.s")
+            collectible = (EnItem00*)Item_DropCollectible(globalCtx, &spawnPos, 0xE);
```

Comment:
odd to me that this function doesnt just return an EnItem00, which would avoid the casting. 
@Roman971 thoughts?

Reply:
It should return a `EnItem00*`, but it doesn't currently because it's exposed in functions.h and the EnItem00 struct isn't exposed globally.

Reply:
oooo right
sirius when you move enitem00 to z64_actor can you change this as well?

Reply:
Yep! Done.